x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/InitializePolicyContextStep.java

```diff
@@ -25,8 +27,11 @@
     public static final StepKey KEY = new StepKey(INITIALIZATION_PHASE, "init", "init");
     private static final Logger logger = LogManager.getLogger(InitializePolicyContextStep.class);
 
-    InitializePolicyContextStep(Step.StepKey key, StepKey nextStepKey) {
+    private final String policy;
+
+    InitializePolicyContextStep(String policy, Step.StepKey key, StepKey nextStepKey) {
         super(key, nextStepKey);
+        this.policy = policy;
```

Comment:
We don't need to pass the policy in and store it in the step, we can get it directly out of the index metadata (the `index.lifecycle.name` setting)

Reply:
Ah yes, great point!